### DISCUSSION TITLE
When should you use useState vs. useRef? — Jagad Yudha Awali

### DISCUSSION THREAD

**Discussion by giscus** [jagadyudha/jagad.dev#94] [Announcements]
# When should you use useState vs. useRef? — Jagad Yudha Awali

Understanding the similarities and differences between useState and useRef

https://jagad.dev/posts/when-should-you-use-usestate-vs-useref

**Comment by KhalfaniW** [comment]
I don't understand the need to avoid unnecessary renders unless  there is a performance issue.

Using useState with forms is standard practice.  Optimizations like these will confuse other developers, so it is best to follow standard designs.
